Sources/Operators/FlatMapFirst.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  FlatMapFirst.swift
+//  CombineExt
+//
+//  Created by Martin Troup on 22/03/2022.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
+
+#if canImport(Combine)
+import Combine
+import Foundation
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+public extension Publisher {
+    /// The operator is a special case of `flatMap` operator.
+    ///
+    /// Like `flatMapLatest`, it only allows one inner publisher at a time. Unlike `flatMapLatest`, it will not cancel an ongoing inner publisher.
+    /// Instead it ignores events from the source until the inner publisher is done. It creates another inner publisher only when the previous one is done.
+    ///
+    /// - Returns: A publisher emitting the values of a single inner publisher at a time (until the inner publisher finishes).
+    func flatMapFirst<P: Publisher>(
+        _ transform: @escaping (Output) -> P
+    ) -> Publishers.FlatMap<Publishers.HandleEvents<P>, Publishers.Filter<Self>>
+    where Self.Failure == P.Failure {
+        var isRunning = false
+        let lock = NSRecursiveLock()
+
+        func set(isRunning newValue: Bool) {
+            defer { lock.unlock() }
+            lock.lock()
+
+            isRunning = newValue
+        }
+
+        return self
```

Comment:
Interesting idea for an implementation! Also seems like a good test suite to cover it.
Can you remove the explicit self here?

Reply:
Sure, it should be gone...